library/net.c

```diff
@@ -203,7 +207,11 @@ int mbedtls_net_bind( mbedtls_net_context *ctx, const char *bind_ip, const char
 
         n = 1;
         if( setsockopt( ctx->fd, SOL_SOCKET, SO_REUSEADDR,
+#if VXWORKS
+                        (char *) &n, sizeof( n ) ) != 0 )
+#else
```

Comment:
No need for ifdefs on any of these three casts in this file: the cast to a (non-const) char* should be legal on all platforms, even though it's not required with modern const-correct headers.